### PR TITLE
Add transaction signing capabilities

### PR DESCRIPTION
**Pull request by alisina-nl**
This PR adds the following features to Ethers:

- A `Signer` behaviour which has mandatory callbacks for listing accounts and signing transactions.
- Two built-in signer implementations:
  - `Signer.Local`: A local signer which signs transactions and lists accounts (a single account) using a 256bit private key.
  - `Signer.JsonRPC`: A signer which delegates the transaction signing to a RPC server (like web3siger or geth) using the `eth_signTransaction` and `eth_accounts` functions.
- Transaction struct with support for encoding EIP-1559 and Legacy transactions.
- A new `Ethers.sign_transaction/2` function which returns signed payload of a transaction.
- Accepting `signer` and `signer_opts` options in `Ethers.send/2` and `Ethers.deploy/2` functions.
- Updated documentation (+ README.md) with the new signer.

Closes fortelabsinc/elixir_ethers#1

### CODE REVIEW

**Comment by wchenNL** [comment]
while testing with this branch found a few very minor things which may not be related to this PR:
- `default_address` is required otherwise a `{:error, :no_to_address}` is returned, so I guess it's not really optional?
- extra comma in the readme at this line
```default_signer_opts: [] # Defaults to: []```

File: lib/ethers/signer/local.ex [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      end
+    end
+  end
+
+  defp priv_to_pub(private_key), do: secp256k1_module().create_public_key(private_key)
+
+  defp private_key(opts) do
+    case Keyword.get(opts, :private_key) do
+      <<key::binary-32>> -> {:ok, key}
+      <<"0x", _::binary-64>> = key -> Ethers.Utils.hex_decode(key)
```
Comment (wchenNL) [inline lib/ethers/signer/local.ex:80]: We may want to support private key that has `0x` stripped out too? I saw private keys like that quite often. So users do not need to prepend `0x` on their end. Maybe also add some description on types of private key we support here.

Comment (alisinabh) [reply lib/ethers/signer/local.ex:82]: Good point. Addressed in 86ce06b

**Comment by alisinabh** [comment]
> * `default_address` is required otherwise a `{:error, :no_to_address}` is returned, so I guess it's not really optional?

This is true. If you don't provide the default address you MUST specify `to` address when calling or sending a transaction. In that case you can do it like this (in the second argument of send/call function):

```elixir
MyERC20Token.name()
|> Ethers.call(to: "0xContract Address Here")
```

If this is not well communicated in the docs maybe we should improve them.

**Review by wchenNL — APPROVED** [review]
LGTM!

**Pull request merged**